common/util/cmd_positional_arguments.h

```diff
@@ -0,0 +1,67 @@
+// Copyright 2017-2020 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef VERIBLE_COMMON_UTIL_CMD_POSITIONAL_ARGUMENTS_H_
+#define VERIBLE_COMMON_UTIL_CMD_POSITIONAL_ARGUMENTS_H_
+
+#include <string>
+#include <vector>
+
+#include "absl/status/status.h"
+#include "absl/strings/string_view.h"
+
+namespace verible {
+
+class CmdPositionalArguments {
+ public:
+  explicit CmdPositionalArguments(const std::vector<char *> &args)
+      : all_args_(args){};
+
+  // Main function that parses arguments.
+  absl::Status ParseArgs();
+
+  // Gets include directories arguments.
+  std::vector<absl::string_view> GetIncludeDirs();
+
+  // Gets macro defines arguments.
+  std::vector<std::pair<absl::string_view, absl::string_view>> GetDefines();
+
+  // Gets SV files arguments.
+  std::vector<absl::string_view> GetFiles();
+
+ private:
+  // Sets a include directory argument.
+  absl::Status SetIncludeDir(absl::string_view include_dir);
+
+  // Sets a define arguments.
+  absl::Status SetDefine(
+      std::pair<absl::string_view, absl::string_view> define);
+
+  // Sets a file argument.
+  absl::Status SetFile(absl::string_view file);
+
+  std::vector<char *>
+      all_args_;  // contains all arguments (tool's name is included).
+  std::vector<absl::string_view>
+      include_dirs_;  // contains all arugments that follows +incdir+<dir>.
+  std::vector<absl::string_view>
+      files_;  // contains all SV files passed to the tool.
+  std::vector<std::pair<absl::string_view, absl::string_view>>
```

Comment:
instead of a `std::pair<>`. make this a struct with

```c++
struct TextMacroDefinition {
   absl::string_view  name;
   absl::string_view value;
}
```
(and this will be a member `std::vector<TextMacroDefinition> macros` in the `struct FileList` then).

Also, given that the other members in FileList actually store `std::string`, we should do the same in `TextMacroDefinition`, thus having the concern about the std::vector<char*>` having to outlive the class eliminated.

Note, we also already have a [`struct MacroDefinition`](https://github.com/chipsalliance/verible/blob/6b2724257d54225491ee48214fa08f4ddfb4d29c/common/text/macro_definition.h#L76) which internally uses Tokens to refer to values.These tokens come from parsed files so it is more complicated. So for now it would be ok to just come up with our own struct and then later at some point consolidate. So when you declare the proposed `struct TextMacroDefinition`, add a comment pointing to `MacroDefinition` as future work.